fpmsyncd/routesync.cpp

```diff
@@ -886,6 +996,106 @@ bool RouteSync::getIfName(int if_index, char *if_name, size_t name_len)
+ * @arg mpls_list     (output) comma-separated list of NH MPLS info
+ * @arg intf_list     (output) comma-separated list of NH interfaces
+ *
+ * Return void
+ */
+void RouteSync::getNextHopList(struct rtnl_route *route_obj, string& gw_list,
+                               string& mpls_list, string& intf_list)
+{
+    bool mpls_found = false;
+
+    for (int i = 0; i < rtnl_route_get_nnexthops(route_obj); i++)
+    {
+        struct rtnl_nexthop *nexthop = rtnl_route_nexthop_n(route_obj, i);
+        struct nl_addr *addr = NULL;
+
+        /* RTA_GATEWAY is NH gateway info for IP routes only */
+        if ((addr = rtnl_route_nh_get_gateway(nexthop)))
+        {
+            char gw_ip[MAX_ADDR_SIZE + 1] = {0};
+            nl_addr2str(addr, gw_ip, MAX_ADDR_SIZE);
+            gw_list += gw_ip;
+
+            /* LWTUNNEL_ENCAP_MPLS RTA_DST is MPLS NH label stack for IP routes only */
+            if ((addr = rtnl_route_nh_get_encap_mpls_dst(nexthop)))
+            {
+                char labelstack[MAX_ADDR_SIZE + 1] = {0};
+                nl_addr2str(addr, labelstack, MAX_ADDR_SIZE);
+                mpls_list += string("push+");
+                mpls_list += labelstack;
+                mpls_found = true;
+            }
+            /* Filler for proper parsing in routeorch */
+            else
+            {
+                mpls_list += string("na");
+            }
+        }
+        /* RTA_VIA is NH gateway info for MPLS routes only */
+        else if ((addr = rtnl_route_nh_get_via(nexthop)))
+        {
+            char gw_ip[MAX_ADDR_SIZE + 1] = {0};
+            nl_addr2str(addr, gw_ip, MAX_ADDR_SIZE);
+            gw_list += gw_ip;
+
+            mpls_found = true;
+            mpls_list += string("swap+");
+
+            /* RTA_NEWDST is MPLS NH label stack for MPLS routes only */
+            if ((addr = rtnl_route_nh_get_newdst(nexthop)))
+            {
+                char labelstack[MAX_ADDR_SIZE + 1] = {0};
+                nl_addr2str(addr, labelstack, MAX_ADDR_SIZE);
+                mpls_list += labelstack;
+            }
+        }
+        else
+        {
+            if (rtnl_route_get_family(route_obj) == AF_INET6)
+            {
+                gw_list += "::";
+            }
+            /* for MPLS route, use IPv4 as default gateway. */
+            else
+            {
+                gw_list += "0.0.0.0";
+            }
+            mpls_list += string("na");
+        }
+
+        /* Get the ID of next hop interface */
+        unsigned if_index = rtnl_route_nh_get_ifindex(nexthop);
+        char if_name[IFNAMSIZ] = "0";
+        if (getIfName(if_index, if_name, IFNAMSIZ))
+        {
+            intf_list += if_name;
+        }
+        /* If we cannot get the interface name */
+        else
+        {
+            intf_list += "unknown";
+        }
+
+        if (i + 1 < rtnl_route_get_nnexthops(route_obj))
+        {
+            gw_list += string(",");
```

Comment:
How about using "NHG_DELIMITER" instead of using ","

Reply:
@smaheshm Done